### PR TITLE
Makes the jukebox less terrible

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: The jukebox song will no longer restart when you leave and re-enter the room.
fix: The jukebox will no longer freeze the game.
/:cl:

### CODE REVIEW

**Comment by KorPhaeron** [comment]
This seems inconsistent at eliminating the lag, but it felt shorter and less frequent I think

Sound files probably just need to be smaller now

**Comment by Aloraydrel** [comment]
Based

**Comment by Dorsisdwarf** [comment]
Is there any way you can fix it being a binary on/off between it going from full loudness to off the instant its more than 7 tiles away? 

Also it takes a second or so to restart when you come back into range, which is weird and disjointed

**Comment by optimumtact** [comment]
the lag comes from the file being streamed to you, it locks up your client.

For people like me who are further away this lag is anywhere upwards of 20 seconds

**Comment by KorPhaeron** [comment]
@optimumtact I'm not at all familiar with goonchat, would sending music from the jukebox through it/stopping music sent by it conflict with admin played youtube links?

**Comment by optimumtact** [comment]
you can add more html5 audio elements.

I'm sure it will be incredible

**Comment by ExcessiveUseOfCobblestone** [comment]
mfw when you leave the area and come back it just reuploads the song to you.

File: code/game/machinery/dance_machine.dm [hunk 1]
```diff
@@ -144,9 +145,11 @@
...
 	active = TRUE
 	update_icon()
-	START_PROCESSING(SSobj, src)
+	playing_song = sound(selection.song_path)
+	playing_song.status = SOUND_STREAM | SOUND_UPDATE
+	playing_song.channel = CHANNEL_JUKEBOX
 	stop = world.time + selection.song_length
-
+	START_PROCESSING(SSobj, src)
 /obj/machinery/jukebox/disco/activate_music()
```
Comment (Cyberboss) [inline code/game/machinery/dance_machine.dm]: Spacing

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request closed without merging**